code/game/machinery/machinery.dm

```diff
@@ -489,8 +489,8 @@ Class Procs:
 	return
 
 /obj/machinery/proc/eject_material_of_type(var/incoming_material) //Used for autolathe, protolathe, mechfab, exofab. Stuff that takes materials, basically.
-	if(LAZYLEN(materials))
-		if(materials[incoming_material])
+	if(!LAZYACCESS(materials[incoming_material]))
+		return
 			var/datum/material/M = get_material_by_name(incoming_material)

```

Comment:
You have to dedent the rest of this block, it's scoped to run after the return right now (i.e., never).

Reply:
Thanks, not sure how that got like that.